src/types/utils.cpp

```diff
@@ -378,71 +112,315 @@ til::color Utils::ColorFromHexString(const std::string_view str)
 // Arguments:
-// - table: a color table with at least 16 entries
+// - wch - Character to convert.
+// - value - receives the int value of the char
 // Return Value:
-// - <none>, throws if the table has less that 16 entries
-void Utils::InitializeCampbellColorTableForConhost(const gsl::span<COLORREF> table)
+// - true iff the character is a hex character.
+bool Utils::HexToUint(const wchar_t wch,
+                      unsigned int& value) noexcept
 {
-    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
-    InitializeCampbellColorTable(table);
-    SwapANSIColorOrderForConhost(table);
+    value = 0;
+    bool success = false;
+    if (wch >= L'0' && wch <= L'9')
+    {
+        value = wch - L'0';
+        success = true;
+    }
+    else if (wch >= L'A' && wch <= L'F')
+    {
+        value = (wch - L'A') + 10;
+        success = true;
+    }
+    else if (wch >= L'a' && wch <= L'f')
+    {
+        value = (wch - L'a') + 10;
+        success = true;
+    }
+    return success;
 }
 
-// Function Description:
-// - modifies in-place the given color table from ANSI (RGB) order to Console order (BRG).
+// Routine Description:
+// - Converts a number string to its equivalent integer value.
 // Arguments:
-// - table: a color table with at least 16 entries
+// - wstr - String to convert.
+// - value - receives the int value of the string
 // Return Value:
-// - <none>, throws if the table has less that 16 entries
-void Utils::SwapANSIColorOrderForConhost(const gsl::span<COLORREF> table)
+// - true iff the string is a number string.
+bool Utils::StringToUint(const std::wstring_view wstr,
+                         unsigned int& value)
 {
-    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
-    std::swap(til::at(table, 1), til::at(table, 4));
-    std::swap(til::at(table, 3), til::at(table, 6));
-    std::swap(til::at(table, 9), til::at(table, 12));
-    std::swap(til::at(table, 11), til::at(table, 14));
+    unsigned int result = 0;
+    size_t current = 0;
+    while (current < wstr.size())
+    {
+        const wchar_t wch = wstr.at(current);
+        if (_isNumber(wch))
+        {
+            result *= 10;
+            result += wch - L'0';
+
+            ++current;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    value = result;
+
+    return true;
 }
 
-// Function Description:
-// - Fill the first 255 entries of a given color table with the default values
-//      of a full 256-color table
+
+// Routine Description:
+// - Split a string into different parts using the delimeter provided.
 // Arguments:
-// - table: a color table with at least 256 entries
+// - wstr - String to split.
+// - delimeter - delimeter to use.
 // Return Value:
-// - <none>, throws if the table has less that 256 entries
-void Utils::Initialize256ColorTable(const gsl::span<COLORREF> table)
+// - a vector containing the result string parts.
+std::vector<std::wstring_view> Utils::SplitString(const std::wstring_view wstr,

```

Comment:
I don't know if this can be simplified any more ...

Reply:
And honestly I cannot believe there's no similar functions in STL. I mean, come on. This makes me miss C# and Java soooo much.